downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md

```diff
@@ -82,12 +82,19 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
 ## Developer tools updates
 
 ### New features
+#### OpenAPI Tool
+Added support to generate Ballerina client and service declarations from Swagger 2.0(i.e. OpenAPI 2.0) definitions
 
 #### Language Server
 
 ### Improvements
 
 #### OpenAPI Tool
+- Added a mapping for the `additionalProperties` property in the OpenAPI object schema. With this support, the generated Ballerina records can be open or closed according to the `additionalProperties` details.
+>**Info:** Thereby, some of your existing generated Ballerina records may change into closed records after regenerating using the Ballerina 2201.4.0 (Swan Lake) version.
+- Improved the OpenAPI schema `nullable:true` property support by providing generated a record field with `nil` as the default value (e.g. : `string name = ()` ), instead of making field both nilable and optional (e.g. : `string? name?;`).
```

Comment:
```suggestion
- Improved support for `nullable:true` property OpenAPI schema, to generate record fields with default values (e.g. `string? name = ();`), instead of making the field both nilable and optional (e.g. `string? name?;`).
```